### PR TITLE
Feature/file upload

### PR DESCRIPTION
**Pull request by Parasuraman-Annadurai**
1. File upload added both task and assessment

### CODE REVIEW

File: src/components/AddBatchModule/AddBatch.jsx [hunk 1]
```diff
@@ -44,7 +44,8 @@ const AddBatch = (props) => {
       ),
       content: "Are you sure you want to Switch this Batch?",
       onOk: () => {
-        navigate(`/batch/${batch.id}/applications`);
+        const navigateUrl = getPermission(user.permissions,"Applicant","read") ? `/batch/${batch?.id}/applications` : `/batch/${batch?.id}/task`
+        navigate(navigateUrl);
         window.location.reload();
```
Comment (madhavakm) [inline src/components/AddBatchModule/AddBatch.jsx]: we are navigating to another page why we need to reload the page again?

File: src/components/AddBatchModule/AddBatch.jsx [hunk 2]
```diff
@@ -237,9 +238,14 @@ const AddBatch = (props) => {
...
           }
-          onClose={()=>{props.onClose() }}
+          onClose={()=>{props.onClose(),
+            setBatchshow(false)
+          }}
           open={props.open}
           placement="left"
+          maskClosable={()=>{
+            setBatchshow(false)
+          }}
```
Comment (madhavakm) [inline src/components/AddBatchModule/AddBatch.jsx]: As per doc in this attr their expecting only boolen but you are providing like a arrow fn is that working?

File: src/components/AssessmentView.jsx [hunk 3]
```diff
@@ -64,16 +66,15 @@ const AssessmentView = ({
...
 }) => {
   const { id: batchId } = useParams();
   const { token, user } = useAuth();
-  const [loading, setLoading] = useState(false);
-  const [isEditing, setIsEditing] = useState(false);
-  const [initialTitle, setInitialTitle] = useState("");
   const [studentScore, setStudentScore] = useState([]);
   const [toggleAssigneeWeightage, setToggleAssigneeWeightage] = useState(
-    type == "task" ? 0 : 1
+    type == "TASK" ? 0 : 1
```
Comment (madhavakm) [inline src/components/AssessmentView.jsx]: maintian this type of constants in another file.

File: src/pages/assessmentModule/AssessmentModule.jsx [hunk 4]
```diff
@@ -36,7 +36,8 @@ const AssessmentModule = ({ type }) => {
   const [formErrors, setFormErrors] = useState({});
   const [weightageErrors, setWeightageErros] = useState({});
   const [assigneeSearch,setAssigneeSearch] = useState("");
-  const [isAssigneeLoading,setIsAssigneeLoading] = useState(false)
+  const [isAssigneeLoading,setIsAssigneeLoading] = useState(false);
+  const [isAssessmentLoading,setIsAssessmentLoading] = useState(false);
```
Comment (madhavakm) [inline src/pages/assessmentModule/AssessmentModule.jsx]: instead using this many usestate, declare a object in usestate try to update key based on need.

File: src/pages/applications/Applications.jsx [hunk 5]
```diff
@@ -526,7 +532,7 @@ const Applications = () => {
                         </div>
                         <div className="application-gender">
                           <p>Gender</p>
-                          <span>-</span>
+                          <span>{application?.gender?.charAt(0).toUpperCase()}{application?.gender?.slice(1).toLowerCase()}</span>
```
Comment (madhavakm) [inline src/pages/applications/Applications.jsx]: make this reusable function, because i have seen lot places uses the same logic.

**Review by madhavakm — CHANGES_REQUESTED** [review]

File: src/pages/applications/Applications.jsx [hunk 6]
```diff
@@ -23,7 +23,7 @@ import useFilter from "../../hooks/useFilter";
 
 import "./scss/css/Applications.css";
 
-import { getPermission } from "../../utils/validate";
+import { getPermission,capitalizeFirstLetter } from "../../utils/validate";
```
Comment (madhavakm) [inline src/pages/applications/Applications.jsx]: Name thefunction name properly.

**Review by madhavakm — CHANGES_REQUESTED** [review]

File: src/pages/assessmentModule/AssessmentModule.jsx (hunk 4 above)
Comment (Parasuraman-Annadurai) [reply src/pages/assessmentModule/AssessmentModule.jsx]: We Need disscussion

File: src/components/AssessmentView.jsx (hunk 3 above)
Comment (Parasuraman-Annadurai) [reply src/components/AssessmentView.jsx]: We Need disscussion

**Pull request closed without merging**